integration_tests/variable_decl_02.py

```diff
@@ -0,0 +1,14 @@
+from ltypes import i32, i64
+
+def f():
+    d: i32 = 0
+    c: i64 = i64(d + 1)
+    b: i64 = i64(d) + c + i64(2)
+    a: i32 = i32(b) + 20
+    print(a, b, c, d)
+    assert d == 0
+    assert c == i64(1)
+    assert b == i64(3)
+    assert a == 23
+
+f()
```

Comment:
This test already works with the current master:
```
$ lpython a.py 
23 3 1 0
```
So that's not a good test for this PR?

Reply:
I didn't try it on `main`. However I would suggest to keep this test as it verifies the changes of this PR as well and its a very small one so shouldn't effect the test timings by much. Originally I wasn't planning to add new tests as existing ones felt sufficient but then I added two new tests anyways to make sure whatever I changes I did work correctly.